### PR TITLE
Use @dataclass for cirq.google.[De]SerializingArg

### PR DESCRIPTION
**Pull request by Strilanc**

### CODE REVIEW

File: cirq/google/op_deserializer.py [hunk 1]
```diff
@@ -16,26 +16,23 @@
...
-            ('constructor_arg_name', str),
-            ('value_func', Optional[Callable[[arg_func_langs.ArgValue], Any]]),
-            ('required', bool),
-        ])):
+@dataclass
+class DeserializingArg:
     """Specification of the arguments to deserialize an argument to a gate.
 
-    Attributes:
+    Args:
```
Comment (Strilanc) [inline cirq/google/op_deserializer.py]: I realize this looks silly. But in my experimentation sphynx only seemed to understand the Args section, not the Attributes section.

Comment (mpharrigan) [reply cirq/google/op_deserializer.py]: you could try digging into the napoleon documentation to see if there's some flag you need to set. IIRC napoleon (used for turning google-style docstrings into sphinx/rst) has a lot of flags you can set

File: cirq/google/op_serializer.py [hunk 2]
```diff
@@ -31,17 +33,11 @@
...
 
-class SerializingArg(
-        NamedTuple(
-            'SerializingArg',
-            [('serialized_name', str),
-             ('serialized_type', Type[arg_func_langs.ArgValue]),
-             ('gate_getter',
-              Union[str, Callable[['cirq.Gate'], arg_func_langs.ArgValue]]),
-             ('required', bool)])):
+@dataclass
```
Comment (mpharrigan) [inline cirq/google/op_serializer.py]: consider `@dataclass(frozen=True)` for immutability and hashability

File: cirq/google/op_deserializer.py (hunk 1 above)
Comment (Strilanc) [reply cirq/google/op_deserializer.py]: Reverted attributes->args change (will address in later PR if needed)

File: cirq/google/op_serializer.py (hunk 2 above)
Comment (Strilanc) [reply cirq/google/op_serializer.py]: Done.

**Pull request merged**